R/intervals_support_funs.R

```diff
@@ -0,0 +1,244 @@
+# Load necessary library
+library(dplyr)
+
+#' Remove specified imputation methods from the intervals in a PKNCAdata object.
+#'
+#' @param data A PKNCAdata object containing the intervals and data components.
+#' @param target_impute A character string specifying the imputation method to be removed.
+#' @param target_params A character vector specifying the parameters to be targeted (optional). If missing, all TRUE in the intervals are taken.
+#' @param target_groups A named list specifying the intervals to be targeted (optional). If missing, all relevant groups are considered.
+#' @return A modified PKNCAdata object with the specified imputation methods removed from the targeted intervals.
+#' @examples
+#' d_conc <- data.frame(
+#'   conc = c(1, 0.6, 0.2, 0.1, 0.9, 0.4, 1.2, 0.8, 0.3, 0.2, 1.1, 0.5),
+#'   time = rep(0:5, 2),
+#'   analyte = rep(c("Analyte1", "Analyte2"), each = 6),
+#'   include_hl = c(FALSE, NA, TRUE, TRUE, TRUE, TRUE, FALSE, NA, TRUE, TRUE, TRUE, TRUE)
+#' )
+#'
+#' d_dose <- data.frame(
+#'   dose = c(100, 200),
+#'   time = c(0, 0),
+#'   treatment = c("A", "B"),
+#'   ID = c(1, 2)
+#' )
+#'
+#' o_conc <- PKNCAconc(d_conc, conc ~ time | analyte, include_half.life = "include_hl")
+#' o_dose <- PKNCAdose(d_dose, dose ~ time | treatment + ID)
+#'
+#' intervals <- data.frame(
+#'   start = c(0, 0, 0),
+#'   end = c(24, 48, Inf),
+#'   half.life = c(TRUE, FALSE, TRUE),
+#'   impute = c("start_conc0,start_predose", "start_predose", "start_conc0"),
+#'   ANALYTE = c("Analyte1", "Analyte2", "Analyte1"),
+#'   ROUTE = c("intravascular", "oral", "intravascular")
+#' )
+#'
+#' o_data <- PKNCAdata(o_conc, o_dose, intervals = intervals)
+#'
+#' # Apply interval_remove_impute function
+#' o_data <- interval_remove_impute(o_data, target_impute = "start_conc0", target_params = c("half.life"), target_groups = list(ANALYTE = "Analyte1", ROUTE = "intravascular"))
+#'
+#' # Print updated intervals
+#' print("Updated intervals:")
+#' print(o_data$intervals)
+#'
+#' @export
+interval_remove_impute <- function(data, target_impute, target_params = NULL, target_groups = NULL) {
```

Comment:
Please make the function S3 generic with methods for `PKNCAdata` and `data.frame`. Then, please make the `PKNCAdata` method use the `data.frame` method.